staging/src/k8s.io/kubectl/pkg/explain/v2/generator.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+
+	sprig "github.com/go-task/slim-sprig"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type TemplateContext struct {
+	GVR       schema.GroupVersionResource
+	Document  map[string]interface{}
+	Recursive bool
+	FieldPath []string
+}
+
+type generator struct {
+	document  map[string]interface{}
+	templates map[string]*template.Template
+}
+
+func NewGenerator(
+	document map[string]interface{},
+) *generator {
+	return &generator{
+		document:  document,
+		templates: make(map[string]*template.Template),
+	}
+}
+
+func (g *generator) AddTemplate(name string, contents string) error {
+	compiled := template.
+		New(name)
+
+	compiled = compiled.
+		Funcs(sprig.HermeticTxtFuncMap())
+
+	_, err := compiled.Parse(contents)
```

Comment:
I can't remember if you changed that, has someone suggested you split these into 3?
```suggestion
	compiled, err := template.New(name).Funcs(sprig.HermeticTxtFuncMap()).Parse(contents)
```

Reply:
fixed